notebooks/05-operations/01-druid-logs-configuration.ipynb

```diff
@@ -0,0 +1,524 @@
+    "\n",
+    "```bash\n",
+    "ls -l\n",
+    "```\n",
+    "\n",
+    "Use other Log4j policies in the usual way. For example, you can adjust the `filePattern` to something like `%d{yyyyMMdd-HH:mm}-%i.log` and then use a [`SizeBasedTriggeringPolicy`](https://logging.apache.org/log4j/2.x/manual/appenders.html#sizebased-triggering-policy) instead of a `TimeBasedTriggeringPolicy` to have log files emitted when a log file hits a particular size, rather than based on the timestamp."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4cab489e-aa10-4482-a62d-3c106bdb06cc",
+   "metadata": {},
+   "source": [
+    "### Log patterns\n",
+    "\n",
+    "The content of log files is specified in the [`PatternLayout`](https://logging.apache.org/log4j/2.x/manual/layouts.html#pattern-layout).\n",
+    "\n",
+    "Run the following command to see the current Log4J configuration for `PatternLayout` for the \"Console\".\n",
+    "\n",
+    "```bash\n",
+    "xmllint -xpath Configuration/Appenders/Console/PatternLayout ~/learn-druid-local/apache-druid/conf/druid/auto/_common/log4j2.xml\n",
+    "```\n",
+    "\n",
+    "Each element refers to a specific piece of recordable information:\n",
+    "\n",
+    "* Timestamp (`%d{ISO8601}`)\n",
+    "* Log level (`%p`)\n",
+    "* Thread name (`[%t]`)\n",
+    "* Logger name (`%c`)\n",
+    "* Message (`%m%n`)\n",
+    "\n",
+    "Run this command to follow along as log entries are added to the process logs for the Coordinator, Overlord, and Broker.\n",
+    "\n",
+    "```bash\n",
+    "multitail -CS log4jnew -du -P a \\\n",
+    "    -f coordinator-overlord.log \\\n",
+    "    -f broker.log\n",
+    "```\n",
+    "\n",
+    "Developers assign different log levels to different entries in the log, indicating how severe a log entry is. In the log you will see entries with these levels:\n",
+    "\n",
+    "* FATAL (system failure)\n",
+    "* ERROR (functional failure)\n",
+    "* WARN (non-fatal issue)\n",
+    "* INFO (notable event)\n",
+    "* DEBUG (program debugging messages)\n",
+    "* TRACE (highly granular execution event)\n",
+    "\n",
+    "The thread name and logger name are helpful for diagnosis, especially for WARN and ERROR conditions.\n",
+    "\n",
+    "Messages describe what has happened, what the state of the process or some significant variable is. These serve not just for diagnosis but for general learning of how the Druid database works.\n",
+    "\n",
+    "In your `multitail` window, press `q` to quit and return to the terminal."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df2f17b3-f359-4df8-a682-7398e7d2adf8",
+   "metadata": {},
+   "source": [
+    "### Log retention\n",
+    "\n",
+    "Since \"FileAppender\" is a [RollingRandomAccessFileAppender](https://logging.apache.org/log4j/log4j-2.4/manual/appenders.html#RollingRandomAccessFileAppender) you can adjust the `DefaultRolloverStrategy` to control retention of logs by adjusting the `Delete` section.\n",
```

Comment:
```suggestion
    "Since \"FileAppender\" is a [RollingRandomAccessFileAppender](https://logging.apache.org/log4j/log4j-2.4/manual/appenders.html#RollingRandomAccessFileAppender), you can adjust `DefaultRolloverStrategy` to control retention of logs by adjusting the `Delete` section.\n",
```